src/DependencyInjection/Compiler/BlockHelperCompilerPass.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\BlockBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\DependencyInjection\Parameter;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * @internal
+ *
+ * NEXT_MAJOR: remove this class
+ */
+final class BlockHelperCompilerPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container): void
+    {
+        /** @var array{by_class: array<class-string, string>, by_type: array<string, string>} $cacheBlocks */
+        $cacheBlocks = $container->getParameter('sonata_block.cache_blocks');
+
+        $hasCacheBlocks = false;
+        foreach ($cacheBlocks as $blocks) {
+            foreach ($blocks as $cacheType) {
+                $hasCacheBlocks = $hasCacheBlocks || $cacheType !== 'sonata.cache.noop';
```

Comment:
@VincentLanglet after your phpstan changes I realized simply checking `count($cacheBlocks` is not enough as the array will indeed always have the two keys.

I now tested it on one of my apps and this check works. By default `sonata.cache.noop` is used which basically disables caching.